src/builders/indexable-link-builder.php

```diff
@@ -729,4 +583,24 @@ protected function update_incoming_links_for_related_indexables( $related_indexa
 			$this->indexable_repository->update_incoming_link_count( $count['target_indexable_id'], $count['incoming'] );
 		}
 	}
+
+	/**
+	 * Updates the image ids when the indexable images are marked as first content image.
+	 *
+	 * @param Indexable         $indexable The indexable to change.
+	 * @param array<string|int> $images    The image array.
+	 *
+	 * @return void
+	 */
+	public function update_first_content_image( Indexable $indexable, array $images ): void {
+		$current_first_content_image = $indexable->open_graph_image;
+
+		$first_content_image_url = \key( $images );
+		$first_content_image_id  = \current( $images );
+
+		if ( $current_first_content_image === $first_content_image_url ) {
+			$indexable->open_graph_image_id = $first_content_image_id;
+			$indexable->twitter_image_id    = $first_content_image_id;
```

Comment:
The fix is not a complete one, because at the beginning of `update_first_content_image()` we check for 
```
$current_first_content_image = $indexable->open_graph_image;
```
so if an OG image is set by the user and the twitter image is left to be `first-content-image`, the `twitter_image_id` will not be updated.